lib/elixir/src/elixir_with.erl

```diff
@@ -57,3 +80,33 @@ build_cases([], DoExpr) ->
   DoExpr;
 build_cases([{_, Meta, _} = Expr | Rest], DoExpr) ->
   {'__block__', Meta, [Expr, build_cases(Rest, DoExpr)]}.
+
+build_else(Meta, WithCases, RefVar, ElseClauses) ->
+  Result = {'result', Meta, nil},
+  CatchAllExist = catch_all_exists(ElseClauses),
+  Clauses = [
+    {'->', Meta, [[{{'^', Meta, [RefVar]}, Result}], Result]}
+    | ElseClauses
+  ] ++ [build_raise(Meta) || not CatchAllExist],
+  {'case', Meta, [WithCases, [{do, Clauses}]]}.
+
+catch_all_exists([{'->', _, [[Match], _]} | ElseClauses]) ->
+  catch_all_exists(Match) orelse catch_all_exists(ElseClauses);
+catch_all_exists([{'->', _, _} | ElseClauses]) ->
+  catch_all_exists(ElseClauses);
+catch_all_exists({'=', _, Expressions}) ->
+  %% We need to check recursive such possible and meaningless conditions too, like
+  %% var1 = var2 -> ..., which are catch all conditions too.
+  lists:all(fun catch_all_exists/1, Expressions);
+catch_all_exists({VariableName, _, NameSpace}) when is_atom(VariableName) andalso is_atom(NameSpace) ->
```

Comment:
Nitpicking: third element is usually called `Context`, also we can rename `VariableName` –> `VarName`. :bowtie: